### PR TITLE
Do not capture System.in all the time

### PR DESCRIPTION
**Pull request by big-guy**
### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [ ] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
- [ ] [Link to Design Spec](https://github.com/gradle/gradle/tree/master/design-docs) for changes that affect more than 1 public API (that is, not in an `internal` package) or updates to > 20 files
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by big-guy** [comment]
@bmuschko I've tried to come up with a integ test for this, but I don't know of a clever way to make the process that we fork read from stdin in the same way it works with an interactive terminal.

Testing this by hand with what @pioterj describes in #2915 works now.

**Comment by bmuschko** [comment]
> I've tried to come up with a integ test for this, but I don't know of a clever way to make the process that we fork read from stdin in the same way it works with an interactive terminal.

Yeah, I can image that writing such a test would be tricky. The build under test could just use `System.console().readLine()`, would run without the daemon but then we'd have to simulate a user input as well.

File: subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/ContinuousBuildActionExecuter.java [hunk 1]
```diff
@@ -70,30 +70,34 @@ public ContinuousBuildActionExecuter(BuildActionExecuter<BuildActionParameters>
...
         if (actionParameters.isContinuous()) {
             SingleMessageLogger.incubatingFeatureUsed("Continuous build");
             DefaultContinuousExecutionGate alwaysOpenExecutionGate = new DefaultContinuousExecutionGate();
+            final CancellableOperationManager cancellableOperationManager = createCancellableOperationManager(actionParameters, cancellationToken);
             return executeMultipleBuilds(action, requestContext, actionParameters, buildSessionScopeServices, cancellableOperationManager, alwaysOpenExecutionGate);
         } else {
             try {
                 return delegate.execute(action, requestContext, actionParameters, buildSessionScopeServices);
             } finally {
+                final CancellableOperationManager cancellableOperationManager = createCancellableOperationManager(actionParameters, cancellationToken);
```
Comment (bmuschko) [inline subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/ContinuousBuildActionExecuter.java]: Can you explain why we also create a `CancellableOperationManager` here?

Comment (big-guy) [reply subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/ContinuousBuildActionExecuter.java]: If we detect running deployments at the end of the build when we are _not_ in continuous build, we enter a sort of 'continuous build' after the first build.  It needs to be cancellable via ctrl+C, so it needs a `CancellableOperationManager`

**Pull request merged**